pyprep/prep_pipeline.py

```diff
@@ -56,14 +56,32 @@ def __init__(self, raw, prep_params, montage, ransac=True, random_state=None):
         self.raw = raw.copy()
         self.ch_names = self.raw.ch_names
         self.raw.set_montage(montage)
-        self.raw.pick_types(eeg=True, eog=False, meg=False)
+        self.raw_non_eeg = self.raw.copy()
```

Comment:
is it necessary to make a full copy here? given that we (used to?) run into RAM issues in pyprep regularly, I'd prefer to not copy if possible.

E.g., *extract* the non-EEG channels without a copy and later add them.

Reply:
What do you mean "extract non-EEG channels without a copy" . Like only make a copy of the channel data rather than the raw object? Besides that I'm not sure how could we extract them.

Although there where many ram issues I think copying the raw object entirely does not compare to them. Usually the ram issues we had were in the other of 9gb or more (though they depend on the user ram capability). Moreover the raw_non_eeg copy a few lines later drops the eeg channels so the consumption gets lower. 

It would be possible to retain the channels without any copy by really verifying how the deeper pyprep code treats the raw object since the to-be-processed channels are passed as a list to some functions but I'm not entirely sure all the code works correctly like that, so thats a risky bet.

Last solution I can come up is to save the non eeg channels on disk to save ram.

In any case I don't think the only-non-eeg copy will be a problem.

Reply:
> Like only make a copy of the non-eeg channel data rather than the raw object?

yeah, not even making a copy - simply splitting the existing raw into two: EEG ... and non-EEG.

But you raise valid points, the copy here is probably not a big problem, especially because the EEG channels are then dropped a few lines below. (as you say)

Reply:
Tests passed, looking forward for the feedback. I'm not sure how to do such splitting but if you got an idea of the implementation I can change it.